client/lib/url/is-external.ts

```diff
@@ -11,10 +11,16 @@ import { isLegacyRoute } from 'lib/route/legacy-routes';
 
 // Base URL used for URL parsing. The WHATWG URL API doesn't support relative
 // URLs, so we always need to provide a base of some sort.
-const BASE_HOSTNAME = 'base.invalid';
+const BASE_HOSTNAME = '__domain__.invalid';
 const BASE_URL = `http://${ BASE_HOSTNAME }`;
 
-export default function isExternal( url: TypedURL ): boolean {
+export default function isExternal( url: URLString ): boolean {
+	// While TypeScript should ensure that `url` really is a string, this method
+	// is still used in a lot of JavaScript contexts, without type checks.
+	if ( ! url && url !== '' ) {
+		return true;
```

Comment:
Note: this matches the previous behaviour of this method.

Reply:
I just pulled out the old code to double-check and yes, that's correct.

I did notice that invalid URLs were throwing exceptions (and they didn't in the original code, interestingly), so I added a `try`/`catch` to match the original behaviour and return `false`.